File: src/main/go/aoc2019/day05.go
package main

import (
	"fmt"
	"time"
)

// Run05 Day 5: Sunny with a Chance of Asteroids
func Run05() {

	t := time.Now()
	day05Part1()
	day05Part2()

	fmt.Printf("=== end (%d ms) ===\n", time.Now().Sub(t).Milliseconds())
}

func day05Part1() {
	a := IntCode(ReadInputStrings("input05.txt")[0])

	comp := IntComputer{}
	comp.init(a)
	comp.provide(1, 2, 3, 4)
	go comp.run()

	// comp.in <- 1

	for !comp.halted() {
	}
}

func day05Part2() {
	a := IntCode(ReadInputStrings("input05.txt")[0])

	comp := IntComputer{}
	comp.init(a)
	comp.provide(5)
	go comp.run()

	// comp.in <- 1

	for !comp.halted() {
	}
}

File: src/main/csharp/aoc2019/Day05.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using aoc;

namespace aoc2019
{
    class Day05 : AocDay
    {

        long[] Program = ReadIntcode("input05.txt");

        public void main()
        {
            Console.WriteLine("=== adventofcode.com/2019/day/5: Sunny with a Chance of Asteroids");
            var t0 = DateTime.Now;

            Console.WriteLine("--- part 1");
            Part1();

            Console.WriteLine("--- part 2");
            Part2();

            Console.WriteLine("=== end ({0} ms)", (DateTime.Now - t0).Milliseconds);
        }

        private void Part1()
        {
            var comp = new IntComputer(Program);
            comp.InputData.Enqueue(1);
            comp.Run();
        }

          private void Part2()
        {
            var comp = new IntComputer(Program);
            comp.InputData.Enqueue(5);
            comp.Run();
        }
    }
}
